src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/CourseActivityBuilder.kt

```diff
@@ -1,47 +1,69 @@
 package uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders
 
-import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.AgencyInternalLocation
-import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.AgencyLocation
+import org.springframework.stereotype.Component
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseActivity
-import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.IEPLevel
-import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.ProgramService
 import java.time.LocalDate
 
+@Component
+class CourseActivityBuilderFactory(private val repository: Repository) {
```

Comment:
Switch builder to a factory so that we can inject the repository and look up any supporting entities. This now means all building work is done exclusively by the builder instead of also in the repository.